File: src/main/java/com/example/Task/management/service/TaskService.java
package com.example.Task.management.service;

import com.example.Task.management.dto.TaskDto;
import com.example.Task.management.dto.TaskDto;

import java.util.List;

public interface TaskService {

    TaskDto addTask(TaskDto taskDto, String assignedBy);

    TaskDto getTask(Long id);
    List<TaskDto> getAllTasks();
//
    TaskDto updateTask(TaskDto taskDto, Long id, String assignedBy);
    void deleteTask(Long id);

    TaskDto completeTask(Long id);

    TaskDto inCompleteTask(Long id);

}

File: Task-ui/src/services/TaskService.js
import axios from "axios";
import { getToken } from "./AuthService";
const BASE_REST_API_URL = 'http://localhost:8080/api/tasks';
axios.interceptors.request.use(function (config) {
    
    config.headers['Authorization'] = getToken();

    return config;
  }, function (error) {
    // Do something with request error
    return Promise.reject(error);
  });
export const getAllTasks = () => axios.get(BASE_REST_API_URL)

export const saveTask = (task) => axios.post(BASE_REST_API_URL, task)

export const getTask = (id) => axios.get(BASE_REST_API_URL + '/' + id)

export const updateTask = (id, task) => axios.put(BASE_REST_API_URL + '/' + id, task)

export const deleteTask = (id) => axios.delete(BASE_REST_API_URL + '/' + id)

export const completeTask = (id) => axios.patch(BASE_REST_API_URL + '/' + id + '/complete')

export const inCompleteTask = (id) => axios.patch(BASE_REST_API_URL + '/' + id + '/in-complete')
export const getAllUsers = () => axios.get('http://localhost:8080/api/auth/users');
export const getTaskById = (id) => axios.get(BASE_REST_API_URL + '/' + id);